bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/RegExpFactory.java

```diff
@@ -207,12 +207,21 @@ private static RegExpAtom translateCharacterClass(RegExpCharacterClass charClass
                 continue;
             }
             if (charAtom != null) {
-                charAtoms[i] = translateCharInCharacterClass((String) charAtom);
+                charAtoms[i] = translateVisitor(charAtom);
             }
         }
         return charClass;
     }
 
+    private static Object translateVisitor(Object node) {
+        if (node instanceof RegExpLiteralCharOrEscape) {
+            return translateLiteralCharOrEscape((RegExpLiteralCharOrEscape) node);
+        } else if (node instanceof String) {
+            return translateCharInCharacterClass((String) node);
+        }
+        return node;
```

Comment:
```suggestion
        } 
        return translateCharInCharacterClass((String) node);
```
The node can only be either `String` or `RegExpLiterlaCharOrEscape`.

Reply:
Until the parsers are synced, let's keep the default case. Can be unseen cases even though grammar wouldn't allow other nodes

Reply:
Why it cannot be an another type node?

Reply:
As per the grammer https://ballerina.io/spec/lang/master/#ReCharSet

Reply:
Need to check once this is synced with the master for cases like:

```
re `[a-z\s]`
```